### DISCUSSION TITLE
Is it possible to access a model's current relationship data outside of prisma.schema?

### DISCUSSION THREAD

**Discussion by emmanuel-londono** [prisma/prisma#21562] [Q&A]
### Question

Is it possible to access a model's current relationships? I would like to make a dynamic upsert/update endpoint (expressJS) that checks to see if the inputted model has relationships. If it is one to many, then it will use upsert instead of update.

I've tried something like this:

### Endpoint

```ts
router.get("/relationships", readFunctions.getModelRelationships, async (req: Request, res: Response) => {
    const data = res.locals.relationships;
    res.send(data);
});
```
### Middleware

```ts
async function getModelRelationships(req: Request, res: Response, next: NextFunction) {
    const url = req.originalUrl;
    try {
        const modelName = await determineModelName(url, prisma);
        if (modelName) {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            const relationships = (prisma as Record<string, any>)[modelName].$relations;
            console.log(relationships);
            const modelInfo = {
                model: modelName,
                relationships: relationships.map((relation: RelationType) => ({
                    name: relation.name,
                    type: relation.type,
                    target: relation.target,
                })),
            };
            res.locals.relationships = modelInfo;
        } else {
            res.locals.relationships = { error: "Model name not found." };
        }
    } catch (err) {
        catchHandler(res, err);
    } finally {
        next();
        await prisma.$disconnect();
    }
}

```

### Sys Info

OS: Windows
Database: PostgreSQL
Node.js version: v16.14.2
RestAPI: Expressjs
Language: Typescript
prisma version: 5.3.1
@prisma/client version: 5.4.2

**Comment by ludralph** [comment]
HI @emmanuel-londono 👋 

There isn't a direct way to access a model's current relationships dynamically, as you've tried in your code. Prisma does not expose a `$relations` property on the model instances. However, you can try using `Prisma.dmmf.datamodel.models` to access a `relationFromField` or `relationToFields` property that could help tell if a relation exists. This doesn't explicitly state if the relation is one to many or one to one.

You may also open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your case for wanting to access a model relationship outside of the Prisma schema.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
